source/AI.cpp

```diff
@@ -1092,8 +1120,13 @@ void AI::AskForHelp(Ship &ship, bool &isStranded, const Ship *flagship)
 			if((helper->IsYours() && !ship.IsYours()) || helper.get() == flagship)
 				continue;
 			// Your escorts should not help each other if already under orders.
-			if(helper->IsYours() && ship.IsYours() && orders.count(helper.get()))
-				continue;
+			auto foundOrders = orders.find(helper.get());
+			if(foundOrders != orders.end())
+			{
+				int helperOrders = foundOrders->second.type;
+				if(helper->IsYours() && ship.IsYours() && helperOrders != Orders::MINE && helperOrders != Orders::HARVEST)
+					continue;
+			}
```

Comment:
Harvest is different behavior than gather or hold position.  Gather (i.e. follow the flagship closely) and hold position (don't move at all and if projectiles hit you move back to original hold position) are both unlike Harvest.

In the case of Harvest, the fleet chases flotsam.  An example of this is, a pirate attacks a ship and that ship discards dozens of tons of cargo.  You can put your fleet into Harvest mode and they will chase all of the discarded cargo.

The same is for asteroid flotsam.  Let's say an NPC ship attacks an asteroid and blows it up; then it jumps out leaving flotsam because it was full.  Your fleet didn't do it so it wouldn't transition to Harvest.  In this case, you can order your fleet into Harvest to go pick up the leftover flotsam.

Reply:
MINE is probably redundant with attack.

Reply:
> MINE is probably redundant with attack.

I think it might be, yes. The only difference being is that `MINE` requires a minable target.

Reply:
I would like to keep it.  Future changes I plan to make will use it more distinctly because chasing an asteroid is different from chasing a ship.

Reply:
@warp-core I found why I have a separate MINE command.  This actually comes from earlier feedback from tehhowch and the high level summary is:

> If your own escorts become disabled your mining fleet should prioritize repairing escorts instead of mining.

If the command was simply "fight my target" then this behavior would affect attacking ships.  i.e. you do not want to stop attacking ships and have your escorts prioritize helping... you want them to keep attacking ships.

However, the reverse logic of what `MINE` provides is: if your fleet is mining and one of your escorts becomes disabled, then your fleet will go assist in its recover instead of ignoring it and continuing to mine.